cmake/modules/SwiftSharedCMakeConfig.cmake

```diff
@@ -42,13 +42,33 @@ endfunction()
 
 function(get_imported_library_prefix outvar target prefix)
   string(FIND "${target}" "${prefix}" ALREADY_HAS_PREFIX)
-  if (ALREADY_HAS_PREFIX)
+  if (ALREADY_HAS_PREFIX EQUAL 0)
     set(${outvar} "" PARENT_SCOPE)
   else()
     set(${outvar} "${prefix}" PARENT_SCOPE)
   endif()
 endfunction()
 
+function(get_imported_library_path outvar target target_type)
+  if ("${target_type}" STREQUAL "EXECUTABLE")
+    set(${outvar} "${LLVM_BINARY_OUTPUT_INTDIR}/${target}" PARENT_SCOPE)
+    return()
+  endif()
+
+  # LLVM compiles module libraries as
+  # ${MODULE_NAME}${CMAKE_SHARED_LIBRARY_SUFFIX} on both Linux and macOS.
```

Comment:
Actually that is incorrect here. On APPLE,  CMAKE_SHARED_MODULE_SUFFIX is set to .so. I am not sure right now why that is broken in this way, but the warning will tell us if things change and we can no longer find it.